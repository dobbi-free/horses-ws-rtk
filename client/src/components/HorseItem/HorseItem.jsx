import React from "react";
import style from "./horse-item.module.css";
import horse from "../../static/horse.svg";

export const HorseItem = ({ name, distance }) => {
  return (
    <>
      <div key={name} className={style.horse}>
        <div
          className={style.horse__item}
          style={{ left: `${distance / 10}%` }}
        >
          <p className={style.horse__name}>{name}</p>
          <img className={style.horse__img} src={horse} alt="#" />
        </div>
      </div>
    </>
  );
};
