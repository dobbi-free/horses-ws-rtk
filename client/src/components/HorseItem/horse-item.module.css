
.horse {
    height: 70px;
    border-bottom: 1px solid;
    width: 100%;
    position: relative;
}

.horse__item {
    position: absolute;
    text-align: center;
}

.horse__name {
    text-align: center;
}

.horse__img {
    width: 20px;
}
