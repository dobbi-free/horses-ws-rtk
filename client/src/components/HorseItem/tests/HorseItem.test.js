import { render } from "@testing-library/react";
import { HorseItem } from "../HorseItem";

jest.mock("react-redux");

describe("HorseItem", () => {
    it("should create HorseItem", () => {
        const component = render(<HorseItem distance={0} name={"Princess Diana"} />);

        expect(component).toMatchSnapshot();
    });
});
