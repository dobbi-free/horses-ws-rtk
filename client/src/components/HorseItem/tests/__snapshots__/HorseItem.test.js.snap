// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorseItem should create HorseItem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="horse"
      >
        <div
          class="horse__item"
          style="left: 0%;"
        >
          <p
            class="horse__name"
          >
            Princess Diana
          </p>
          <img
            alt="#"
            class="horse__img"
            src="horse.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="horse"
    >
      <div
        class="horse__item"
        style="left: 0%;"
      >
        <p
          class="horse__name"
        >
          Princess Diana
        </p>
        <img
          alt="#"
          class="horse__img"
          src="horse.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HorseList should create HorseItem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="horse"
      >
        <div
          class="horse__item"
          style="left: 0%;"
        >
          <p
            class="horse__name"
          >
            Princess Diana
          </p>
          <img
            alt="#"
            class="horse__img"
            src="horse.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="horse"
    >
      <div
        class="horse__item"
        style="left: 0%;"
      >
        <p
          class="horse__name"
        >
          Princess Diana
        </p>
        <img
          alt="#"
          class="horse__img"
          src="horse.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
